openscap_report/scap_results_parser/parsers/oval_test_info_parser.py

```diff
@@ -53,115 +60,30 @@ def _get_objects_by_id(self):
-        return OvalObject(**object_dict)
-
-    def _transform_tuples_to_dict(self, array):
-        out_dict = {}
-        for key, value in array:
-            if key in out_dict:
-                key = self._get_unique_key(key)
-            out_dict[key] = value
-        return out_dict
-
-    def _get_object_items(self, xml_object, xml_collected_object):
-        out = []
-        for element in xml_object.iterchildren():
-            id_in_dict = self._get_unique_id_in_dict(element, out)
-            if element.text and element.text.strip():
-                out.append((id_in_dict, element.text))
-            else:
-                out.append((id_in_dict, self._get_ref_var(element, xml_collected_object)))
-        return [self._transform_tuples_to_dict(out)]
-
-    def _get_ref_var(self, element, xml_collected_object):
-        variable_value = ''
-        if self._collected_object_is_not_none_and_contain_var_ref(
-                element, xml_collected_object):
-            var_id = element.attrib.get('var_ref')
-            for item in xml_collected_object:
-                if var_id == item.attrib.get('variable_id'):
-                    variable_value += item.text
-                elif self.get_key_of_xml_element(item) == 'message':
-                    variable_value += self._complete_message(item, var_id) + '<br>'
-        else:
-            variable_value = 'no value'
-        return variable_value
-
-    @staticmethod
-    def _complete_message(item, var_id):
-        if len(item.text) == MAX_MESSAGE_LEN and var_id[:item.text.find('(')] in var_id:
-            return f"{item.text[:item.text.find('(') + 1]}{var_id})"
-        return item.text
-
-    @staticmethod
-    def _collected_object_is_not_none_and_contain_var_ref(element, collected_object):
-        return collected_object is not None and 'var_ref' in element.attrib
-
-    def _get_item(self, item_ref):
-        item = self.system_data.get(item_ref)
-        out = {}
-        for element in item.iterchildren():
-            if element.text and element.text.strip():
-                out[self._get_unique_id_in_dict(element, out)] = element.text
-        return out
+    def _get_states_by_id(self):
+        data = self.oval_definitions.find(
+            ('.//oval-definitions:states'), NAMESPACES)
+        return self._get_data_by_id(data)
 
     def get_test_info(self, test_id):
         test = self.tests[test_id]
         oval_object = None
+        oval_state = None
+
         for item in self.tests[test_id]:
-            object_id = item.attrib.get('object_ref')
+            object_id = item.attrib.get("object_ref")
             if object_id is not None:
-                oval_object = self.get_object(object_id)
+                oval_object = self.objects_parser.get_object(object_id)
+            state_id = item.attrib.get("state_ref")
+            if state_id is not None:
+                oval_state = self.states_parser.get_state(state_id)
```

Comment:
I don't know the details but I'm curious: Why is this a loop? In each iteration it overwrites the oval_object and oval_state. Wouldn't it be better to just pick the last item? And why is the `self.tests[test_id]` an iterable? Does that mean that there are more tests with the same ID?

Reply:
Yes you are right. This for loop is useless. The value contained in `self.tests` is a dictionary where the key is test_id and the value is an XML element. This loop is used to iterate through the children of the test XML element. I will simplify this code to make it more readable and replace the unnecessary loop with xPath.